php/class-deactivation.php

```diff
@@ -205,6 +228,27 @@ public function rest_endpoint( $endpoints ) {
 		return $endpoints;
 	}
 
+	/**
+	 * Uploads the System Report to the Cloud.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function upload_report() {
+		require_once ABSPATH . '/wp-admin/includes/file.php';
+
+		$report = $this->plugin->get_component( 'report' )->get_report_data();
+		$temp   = get_temp_dir() . $report['filename'];
```

Comment:
@pereirinha - Might be a good idea to use `wp_tempnam()` here, so ensure it's writable and to keep VIP approval.

Reply:
@DavidCramer 

As mentioned in our sync, using this function will add a file extension `.tmp`, requiring us to trim it. Otherwise, it will be part of the public_id on Cloudinary. The `wp_tempnam()` is actually a wrapper around `get_temp_dir()` anyways.